lib/charms/opensearch/v0/opensearch_peer_clusters.py

```diff
@@ -88,6 +89,11 @@ def run(self) -> None:
                 Scope.APP, "deployment-description", deployment_desc.to_dict()
             )
 
+        if deployment_desc.start == StartMode.WITH_GENERATED_ROLES:
+            # trigger roles change on the leader, other units will have their peer-rel-changed
+            # event triggered
+            trigger_peer_rel_changed(self._charm, on_other_units=False, on_current_unit=True)
```

Comment:
There is already a "put" above. Is that not enough?

Reply:
it would be but only if `deployment_desc.state.value == State.ACTIVE` - otherwise not